src/bootstrap/lib.rs

```diff
@@ -343,9 +344,18 @@ impl Build {
         // we always try to use git for LLVM builds
         let in_tree_llvm_info = channel::GitInfo::new(false, &src.join("src/llvm-project"));
 
+        let initial_sysroot = config.initial_rustc.parent().unwrap().parent().unwrap();
+        let initial_lld = initial_sysroot
+            .join("lib")
+            .join("rustlib")
+            .join(config.build)
+            .join("bin")
+            .join("rust-lld");
```

Comment:
Can we make this a configurable option like rustc/cargo, with the default being this?

Reply:
You can just configure the linker in that case.

Reply:
Hm, okay. I guess that makes sense. Would that work already? i.e., can you set the linker to the path this generates today? I suspect the answer is no (given the other changes in this PR).


Can we warn somewhere if the linker and this option are set?

Reply:
It's reasonable to use both `linker` and `use-lld` if you're building multiple targets though. I added some comments instead.